src/AxelarGateway.sol

```diff
@@ -178,6 +186,34 @@ abstract contract AxelarGateway is IAxelarGateway, AdminMultisigBase {
         }
     }
 
+    function _mintTokenAndApproveContractCall(
+        string memory symbol,
+        uint256 amount,
+        address contractAddress,
+        bytes32 payloadHash
+    ) internal {
+        if (amount > 0) {
+            _mintToken(symbol, contractAddress, amount);
+            _setContractCallApproved(
+                contractAddress,
+                keccak256(
+                    abi.encode(ContractCallHashKey.WithToken, tokenAddresses(symbol), amount, payloadHash)
```

Comment:
Why is a `ContractCallHashKey` enum even needed if the resulting hashes are already different given that "with token: has 2 additional parameters?